### ISSUE TITLE
"Use values as identifiers" via Wikidata QIDs adds erroneous items

### ISSUE THREAD

**Issue by VojtechDostal** [OpenRefine/OpenRefine#3785]
I have a list of Wikidata QIDs and I want to use them in an import. Thus, I proceed to simple reconciliation via identifier values ("Use values as identifiers"). However, As many as 1 % of QIDs get erroneously paired to a completely different ID. This is hard to spot and leads to nonsense getting imported into Wikidata.

### To Reproduce
1. Download a list of QIDs, eg: https://w.wiki/3ACf (CSV) and import as a new OpenRefine project
2. First column--> Use values as identifiers. 
3. Extract reconciled QID from 1st column via "cell.recon.match.id".
4. compare 1st column values with values in the column created in (3).

### Expected Behavior
I'd like all QIDs to be reconciled without mistake.

### Screenshots

<img width="259" alt="Bez názvu" src="https://user-images.githubusercontent.com/21329813/113614253-8d7a0180-9652-11eb-96d7-e54a643508c4.png">

### Versions
 - Operating System: Windows 10,
 - Browser Version: Newest Chrome
 - OpenRefine: 3.4.1.

### Datasets
Linked from above. But this has happened to me on numerous occassions.

**Comment by gitonthescene** [comment]
I’m not sure what’s causing this result but it’s worth noting that the unresolved example above points to a [non-existent](https://m.wikidata.org/wiki/Special:EntityData/95084491) entity and no longer is in the results of the query you posted.  Do you have an example where this is not the case?  Perhaps the fact that it didn’t auto-reconcile is alerting you to this fact.

**Comment by gitonthescene** [comment]
FWIW, I just downloaded the items from the query you posted and then applied "Use values as identifiers" and then Reconcile -> Facets -> by judgement and all the entries come up as `matched` using OpenRefine 3.4.1 on a Mac 11.1 with Firefox 87.0.

There are ~65,000 QID's returned by that query.  If you do have an example, it would be great if you could make it small enough to be more easily eyeballed.

**Comment by wetneb** [comment]
I tried to reproduce this but all the reconciled cells were associated to the correct id:
![image](https://user-images.githubusercontent.com/309908/113656740-c9d24f80-969c-11eb-9cf4-4300c7cace01.png)

**Comment by VojtechDostal** [comment]
> I’m not sure what’s causing this result but it’s worth noting that the unresolved example above points to a [non-existent](https://m.wikidata.org/wiki/Special:EntityData/95084491) entity and no longer is in the results of the query you posted. Do you have an example where this is not the case? Perhaps the fact that it didn’t auto-reconcile is alerting you to this fact.
> 
> P.S. I agree that offering alternatives feels odd when using “Use values as identifiers”

The item exists, it's available at: https://www.wikidata.org/wiki/Q95084491 (your link just had bad formatting)

I have plenty of examples from today's (different) id-based reconciliation which also produced this mistake (176 mistake, 33695 correct): 

<img width="156" alt="Bez názvu" src="https://user-images.githubusercontent.com/21329813/113663661-b75f1280-96aa-11eb-859d-c72560a26030.png">

**Comment by wetneb** [comment]
I think it would be easier to reproduce this bug if you could point to a static CSV file - the results of the SPARQL query are likely to change often (especially if the bug is coming from redirected/deleted entities which will quickly get cleaned up from the SPARQL endpoint).

**Comment by VojtechDostal** [comment]
> FWIW, I just downloaded the items from the query you posted and then applied "Use values as identifiers" and then Reconcile -> Facets -> by judgement and all the entries come up as `matched` using OpenRefine 3.4.1 on a Mac 11.1 with Firefox 87.0.
> 
> There are ~65,000 QID's returned by that query. If you do have an example, it would be great if you could make it small enough to be more easily eyeballed.

Yes, I also see 100% of lines matched (it's just that some of them are matched incorrectly).
The query is long because the error only occurs in about 1% of the cases... Maybe it only occurs when the job is this big and somehow the tool is overloaded.

**Comment by wetneb** [comment]
Well, next time you see it happen, please keep a copy of the SPARQL results (not query) and upload them to this issue. The "Use values as identifiers" operation does not communicate with the outside world at all, that should normally be enough to reproduce the issue.

**Comment by gitonthescene** [comment]
> The item exists, it's available at: https://www.wikidata.org/wiki/Q95084491 (your link just had bad formatting)

@VojtechDostal - You’re right.  I left off the initial Q.  Sorry for the confusion.  Still, I couldn’t reproduce from the output of the query as I mentioned above.

**Comment by gitonthescene** [comment]
> Yes, I also see 100% of lines matched (it's just that some of them are matched incorrectly).

I’m confused.  The screenshot shows them unmatched.  Is the screenshot from immediately after reconciling or did you take some other action?

**Comment by VojtechDostal** [comment]
> > Yes, I also see 100% of lines matched (it's just that some of them are matched incorrectly).
> 
> I’m confused. The screenshot shows them unmatched. Is the screenshot from immediately after reconciling or did you take some other action?

Sorry for confusion, I manually unmatched some of them to uncover the wrong reconciliation.

@wetneb Will do once I find out something that's reproducible. Currently I suspect that the mistake happenes randomly and it doesn't matter what the specific input data are. If something like that is even possible :)

**Comment by ostephens** [comment]
@VojtechDostal - a couple of questions:

* What is the original content of the QID column in your example? (is it just a Q number or a URL - which is what I get when I use the query you link to)
* When you choose to use the cell content as identifiers - what reconciliation service are you using? (what is the underlying service URL)

Thanks

**Comment by VojtechDostal** [comment]
> @VojtechDostal - a couple of questions:
> 
> * What is the original content of the QID column in your example? (is it just a Q number or a URL - which is what I get when I use the query you link to)
> * When you choose to use the cell content as identifiers - what reconciliation service are you using? (what is the underlying service URL)
> 
> Thanks

The original content is just a QID (Q qith a number). I remove the "https://..." part after I import the CSV)
I use this service: https://wdreconcile.toolforge.org/en/api

thanks :)

**Comment by gitonthescene** [comment]
@ostephens - Are there logs we can check?

**Comment by wetneb** [comment]
When you observe this, if you could simply export your project as an OpenRefine archive, it would really help.

**Comment by VojtechDostal** [comment]
I think I found a related issue. @wetneb Would you mind trying to reproduce it? I'd like to know if I am going crazy or if this is happening to others too :-D This issue is not about wrong reconciliation but somehow messing up between lines.

Steps to reproduce:

1) download my project at https://drive.google.com/file/d/18wH9Uzl4ZNB08cMR7kUt8sFX0mT_2OXT/view?usp=sharing 
and import it into OpenRefine.

2) Import the following Wikidata schema:
https://drive.google.com/file/d/1Sh3QoePbHCqx0qQvfYRGj6zKhYWkwuSV/view?usp=sharing

The values in column "item" are unique, so each item should get one value in Wikidata and not more.

3) Go to tab "issues":

![image](https://user-images.githubusercontent.com/21329813/132831625-7a59e008-3985-4168-81b0-b8fc86bcafa1.png)

Do you also see the "Czech municipality ID (P7606) added more than once on the same item." issue?

thanks

**Comment by wetneb** [comment]
Thanks I can reproduce it too! It is indeed a very annoying problem and I am not entirely sure where it is coming from. I suspect it could be due to the random generation of the internal recon object ids which could sometimes clash (creating two different recon objects with the same id within the same project).

**Comment by VojtechDostal** [comment]
Thank you. Some of my "wrong reconciliations" (reported above) have likely in fact been caused by this and not by actually wrong QID written in the line.
